### PR TITLE
from smart chain issuing pallet

### PR DESCRIPTION
**Pull request by xiaoch05**

### CODE REVIEW

**Review by hackfisher — APPROVED** [review]

**Comment by hackfisher** [comment]
@xiaoch05 Please fix the conflicts

File: runtime/common/src/helixbridge/mod.rs [hunk 1]
```diff
@@ -290,6 +300,131 @@ pub mod pallet {
...
+			spec_version: u32,
+			weight: u64,
+			gas_limit: u128,
+			failure_nonce: MessageNonce,
+			#[pallet::compact] fee: RingBalance<T>,
+		) -> DispatchResultWithPostInfo {
+			let user = ensure_signed(origin)?;
+
+			// Make sure the user's balance is enough to lock
+			ensure!(T::RingCurrency::free_balance(&user) > fee, <Error<T>>::InsufficientBalance);
```
Comment (acgxv) [inline runtime/common/src/helixbridge/mod.rs:384]: I think this check is unnecessary.

File: runtime/common/src/helixbridge/mod.rs [hunk 2]
```diff
@@ -402,39 +543,72 @@ pub mod pallet {
...
-							ExistenceRequirement::KeepAlive,
-						);
-					} else {
-						let _ = T::RingCurrency::withdraw(
-							&Self::pallet_account_id(),
-							value,
-							WithdrawReasons::TRANSFER,
-							ExistenceRequirement::AllowDeath,
-						);
+		pub fn prun_message(
```
Comment (acgxv) [inline runtime/common/src/helixbridge/mod.rs:547]: prune?

**Comment by acgxv** [comment]
No migration needed?

**Comment by xiaoch05** [comment]
> No migration needed?

yes

**Pull request merged**